crates/topos-tce-storage/src/validator/mod.rs

```diff
@@ -467,12 +505,19 @@ impl WriteStore for ValidatorStore {
             _ = self.pending_tables.pending_pool.delete(&pending_id);
         }
 
-        if let Ok(Some(certificate)) = self
+        if let Ok(Some(next_certificate)) = self
             .pending_tables
             .precedence_pool
             .get(&certificate.certificate.id)
         {
-            self.insert_pending_certificate(&certificate)?;
+            debug!(
+                "Delivered certificate {} unlocks {} for broadcast",
+                certificate.certificate.id, next_certificate.id
+            );
+            self.insert_pending_certificate(&next_certificate)?;
```

Comment:
Why would we insert it as pending here?

Reply:
A `certificate` that waits for its `parent` to be delivered is stored in the `precedence_pool` under `parent_id => certificate`. When a `certificate` is delivered, we check if we have a `child` waiting in the precedence pool, if yes, we put this child in the `pending` pool in order to be delivered.